portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -113,6 +120,12 @@ private Entity createCommentEntity(String newComment) {
 
     Timestamp timestamp = new Timestamp(System.currentTimeMillis());
     commentEntity.setProperty("timestamp", timestamp.getTime());
+
+    UserService userService = UserServiceFactory.getUserService();
+
+    String userEmail = userService.getCurrentUser().getEmail();
```

Comment:
Don't worry about this for this PR, but for the future it will make your code easier to maintain, reuse, and test if you were to provide the email address as an argument to this function, e.g. `private Entity createCommentEntity(String newComment, String email) {...}`, and then pass the email details to it.

As this stands now, this function is retrieving the current user as well as creating a comment entity.  So it needs to have a working `UserService` available to retrieve the user - i.e. you have added a hard-coded dependency on the `UserService` for this function.

This isn't a big deal so don't worry too much for now - when you get to the unit testing section later on it might become clear why having these dependencies mixed up in your business logic is annoying/problematic :-)  Obviously somewhere in your code you will need to use `UserService`, but it is generally best to try and abstract these things away as early in your code's execution path as possible/sensible (and ideally only once) - e.g. when the request first comes in you might want to extract the user's login details in doGet()/doPost() and store the details in a variable for use elsewhere later on in your code.

Reply:
Thank you for the notice, I changed the function .